modules/generic-scala3/build.sbt

```
@@ -0,0 +1,3 @@
+import Dependencies.Version._
+
```

Even though you specified only Scala 3 in `crossScalaVersions` the main `scalaVersion` is still set to the default value (2.12.x). The module only really supports Scala 3, so I think it's better to override it here.

```suggestion

scalaVersion := scala3

```

---

I tried that, but it broke the build. Let me check again...

---

Yes. It breaks the `2.12` build, see: https://github.com/grzegorz-bielski/pureconfig/actions/runs/8473601089/job/23218298773#step:5:264

---

Apparently this is an issue with strict aggregation in PureConfig's build. Trying to run `++ 2.12.19 compile` ends up attempting to compile `generic-scala3` since 2.12.19 is included in `crossScalaVersions` of the root project (and the root project aggregates `generic-scala3`). I think #1643 should fix that.

---

@jcazevedo  yeah, looks like this fixes it. Thx!